server/src/main/java/org/cloudfoundry/identity/uaa/provider/oauth/ExternalOAuthProviderConfigurator.java

```diff
@@ -179,6 +181,30 @@ public List<IdentityProvider> retrieveActive(String zoneId) {
         return retrieveAll(true, zoneId);
     }
 
+    @Override
+    public List<IdentityProvider> retrieveActiveByType(final String type, final String zoneId) {
+        if (!OAUTH20.equals(type) && !OIDC10.equals(type)) {
+            return emptyList();
+        }
+        return providerProvisioning.retrieveActiveByType(type, zoneId).stream()
+                .map(idp -> {
+                    if (idp.getType().equals(OIDC10)) {
```

Comment:
very similar to retrievAll, can we combine it. ?
maybe with type = 1 or type = 2.
The index should cover it

Reply:
fixed with commit 8b29079